exception on grid labels click in 4.1.1a

**Operating system**:                       win 10
**wxPython version & source**:          wxPython-4.1.1a1.dev5032+4402d875-cp37-cp37m-win_amd64
**Python version & source**:            3.7, stock

**Description of the problem**:

In my app is systematic to get the following exception with a single click on a grid column label while in the demo GridDragable.py you should repeatedly click with left and right mouse button on a label to get it.

---------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\user1\envs\af4137\lib\site-packages\wx\lib\gridmovers.py", line 327, in OnPress
    self.lwin.CaptureMouse()
wx._core.wxAssertionError: C++ assertion "!wxMouseCapture::IsInCaptureStack(this)" failed at ..\..\src\common\wincmn.cpp(3326) in wxWindowBase::CaptureMouse(): Recapturing the mouse in the same window?

The problem doesn't occur with wxpython4.1.0
Could it eventually be related to this?
https://github.com/wxWidgets/wxWidgets/commit/408ebfd253144ef3535932314a42fe34256d4a56

Is an exception that could be avoided with changes in gridmovers code?

Marco

--- a/wx/lib/gridmovers.py
+++ b/wx/lib/gridmovers.py
@@ -324,12 +324,14 @@ class GridColMover(wx.EvtHandler):
         colImg = self._CaptureImage(rect)
         self.colWin = ColDragWindow(self.grid,colImg,col)
         self.colWin.Show(False)
-        self.lwin.CaptureMouse()
+        if wx.Window.GetCapture() is not None:
+            self.lwin.CaptureMouse()
         evt.Skip()
 
     def OnRelease(self,evt):
         if self.isDragging:
-            self.lwin.ReleaseMouse()
+            if self.lwin.HasCapture():
+                self.lwin.ReleaseMouse()
             self.colWin.Show(False)
             self.isDragging = False
 
@@ -465,12 +467,14 @@ class GridRowMover(wx.EvtHandler):
         rowImg = self._CaptureImage(rect)
         self.rowWin = RowDragWindow(self.grid,rowImg,row)
         self.rowWin.Show(False)
-        self.lwin.CaptureMouse()
+        if wx.Window.GetCapture() is not None:
+            self.lwin.CaptureMouse()
         evt.Skip()
 
     def OnRelease(self,evt):
         if self.isDragging:
-            self.lwin.ReleaseMouse()
+            if self.lwin.HasCapture():
+                self.lwin.ReleaseMouse()
             self.rowWin.Show(False)
             self.isDragging = False
 

